go/racy-append-to-slice.yml

```diff
@@ -13,23 +13,57 @@ rules:
             ...
             go func(...) {
               ...
-              $SLICE = append($SLICE, $ITEM)
+            }(...)
+            ...
+          }
+      - pattern-inside: |
+          var $SLICE []*$TYPE
+          ...
+          for ... {
+            ...
+            go func(...) {
               ...
             }(...)
             ...
           }
+      - pattern-inside: |
```

Comment:
Does this really cause a race condition? As I understand the pattern it means we have a single go routine writing synchronously to the slice.

Reply:
yes, see https://go.dev/play/p/p0FkpAb2MoN. Two issues there, 1) memory aliasing of `$SLICE` and 2) because `$SLICE` is a slice of pointers the `$ITEM` is likely to escape to the heap. The race occurs as now the memory pointed by `$ITEM` might not change in time before the `append` occurs.

Reply:
I _think_ this issue is slightly orthogonal. As I understand it the issue occurs because we are printing the value at `&i` (in `recordLogs`) simultaneously as we are incrementing the value of `i` (in `main`). There is no issue writing the individual addresses to `r`. Would you say that is correct? If that is the case I think this merits its own rule.